taxcalc/simpletaxio.py

```diff
@@ -0,0 +1,649 @@
+"""
+OSPC taxcalc package simple tax input-output class.
+"""
+# CODING-STYLE CHECKS:
+# pep8 --ignore=E402 simpletaxio.py
+# pylint --disable=locally-disabled simpletaxio.py
+
+import os
+import sys
+import json
+import re
+import numpy as np
+import pandas as pd
+from .policy import Policy
+from .records import Records
+from .calculate import Calculator
+
+
+class SimpleTaxIO(object):
+    """
+    Constructor for the simple tax input-output class.
+
+    Parameters
+    ----------
+    input_filename: string
+        name of required INPUT file.
+
+    reform_filename: string or None
+        name of optional REFORM file with None implying current-law policy.
+
+    Raises
+    ------
+    ValueError:
+        if file with input_filename does not exist.
+        if earliest INPUT year before simtax start year.
+        if latest INPUT year after simtax end year.
+
+    Returns
+    -------
+    class instance: SimpleTaxIO
+    """
+
+    def __init__(self, input_filename, reform_filename):
+        """
+        SimpleTaxIO class constructor.
+        """
+        # construct output_filename and delete old output file if it exists
+        self._output_filename = '{}.out-simtax'.format(input_filename)
+        if os.path.isfile(self._output_filename):
+            os.remove(self._output_filename)
+        # check for existence of file named input_filename
+        if not os.path.isfile(input_filename):
+            msg = 'INPUT file named {} could not be found'
+            raise ValueError(msg.format(input_filename))
+        # read input file contents into self._input dictionary
+        self._read_input(input_filename)
+        self._policy = Policy()
+        # implement reform if reform file is specified
+        if reform_filename:
+            reform = SimpleTaxIO.dictionary_from_reform_file(reform_filename)
+            self._policy.implement_reform(reform)
+        # validate input variable values
+        self._validate_input()
+        self._calc = self._calc_object()
+        self._output = {}
+
+    def calculate(self, calc_marginal_tax_rates=False, write_output_file=True):
+        """
+        Calculate taxes for all INPUT lines and write OUTPUT to file.
+
+        Parameters
+        ----------
+        calc_marginal_tax_rates: boolean
+
+        write_output_file: boolean
+
+        Returns
+        -------
+        nothing: void
+        """
+        # loop through self._year_set doing tax calculations and saving output
+        for calcyear in self._year_set:
+            if calcyear != self._calc.policy.current_year:
+                self._calc.policy.set_year(calcyear)
+            self._calc.calc_all()
+            calc_records_tax_year = getattr(self._calc.records, 'FLPDYR')
```

Comment:
Comment from @martinholmer :
   "This is probably because of the indirect way Records members are specified. So, everything in your comment is true, except you are not using pylint, so you don't see the same picture I'm looking at. What I'm doing is correct, right? It just seems indirect to you, right? Or am I missing something else that is important? In other words, is there anything here other the style preferences?"

The code produces the correct result, but it would cause anyone looking at it to say, 'wait, why do it that way?' and then they would wonder if they were missing something. There's no default value added to the `getattr` call, and `FLPDYR` is a string literal and valid Python identifier. The only way this makes sense to me is because you have provided the answer - to silence a warning from pylint. I don't think the best approach is 'get no output from pylint' and I think this is a great example of why it is not always best to do whatever is necessary to get rid of all pylint warnings. pylint is a helpful tool, but it's not flawless.